### ISSUE TITLE
Editor Customization: Add OpenTournament PlayerStart

### ISSUE THREAD

**Issue by wail** [OpenTournament/OpenTournament#354]
**Task Overview**

OpenTournament PlayerStarts have some logic which is intended to help ExperienceMode logic determine valid PlayerStarts successfully (e.g. Team-based Tagging, Claiming to prevent SpawnFrags, etc).

OpenTournament levels should utilize our project's PlayerStart actors. We want to expose our PlayerStart classes to the Editor for usage by Designers.

**Comment by wail** [comment]
```
To modify the Unreal Engine Editor's "Place Actors" window to add a new class of actors that can be placed into the level, you'll need to extend the editor functionality using C++ or Blueprints. Here’s a step-by-step guide on how to do this using C++:

### 1. Create Your Custom Actor Class
First, create the custom actor class that you want to be available in the "Place Actors" window:

1. **Create a New C++ Class**:
   - In Unreal Engine, go to `File` > `New C++ Class`.
   - Choose `Actor` as the parent class.
   - Name your class (e.g., `MyCustomActor`), and click `Create Class`.

2. **Implement the Actor Class**:
   - Open the newly created `MyCustomActor.cpp` and `MyCustomActor.h` files.
   - Implement any custom behavior or properties your actor needs.

### 2. Create a Custom Actor Category in the "Place Actors" Panel
Now, you'll need to extend the Unreal Engine Editor to add a new category in the "Place Actors" panel and include your custom actor in it.

1. **Extend the Editor Module**:
   - In your project, navigate to the `Source` folder.
   - Locate or create an Editor module (if it doesn't exist, you'll need to create one).
   - Open the module's `.Build.cs` file (e.g., `MyProjectEditor.Build.cs`).
   - Add dependencies to the `UnrealEd` and `LevelEditor` modules:

     ```csharp
     PublicDependencyModuleNames.AddRange(new string[] { "UnrealEd", "LevelEditor" });
     ```

2. **Register the Actor in the "Place Actors" Panel**:
   - In your Editor module, implement a custom class to register your actor in the "Place Actors" panel. You can do this in the `StartupModule` function.

   - Create a custom class (e.g., `MyCustomActorsModule`) and implement the registration:

     ```cpp
     #include "MyCustomActorsModule.h"
     #include "LevelEditor.h"
     #include "IPlacementModeModule.h"
     #include "PlacementCategoryInfo.h"
     #include "MyCustomActor.h"

     void FMyCustomActorsModule::StartupModule()
     {
         // Define a custom category
         FPlacementCategoryInfo MyCustomCategory(
             FText::FromString("My Custom Actors"),
             "MyCustomActors",       // Unique identifier for the category
             TEXT("PMActor"),        // Icon Style
             30                      // Sort Order
         );

         // Register the category with the Placement Mode module
         IPlacementModeModule::Get().RegisterPlacementCategory(MyCustomCategory);

         // Register the custom actor class in the custom category
         IPlacementModeModule::Get().RegisterPlaceableItem(
             MyCustomCategory.UniqueHandle,
             MakeShareable(new FPlaceableItem(
                 *UActorFactory::StaticClass(),
                 FAssetData(AMyCustomActor::StaticClass())
             ))
         );
     }

     void FMyCustomActorsModule::ShutdownModule()
     {
         // Unregister the custom category (optional cleanup)
         IPlacementModeModule::Get().UnregisterPlacementCategory("MyCustomActors");
     }
     ```

   - Don't forget to call `IMPLEMENT_MODULE` for your module in the `MyCustomActorsModule.cpp` file:

     ```cpp
     IMPLEMENT_MODULE(FMyCustomActorsModule, MyCustomActors);
     ```

### 3. Compile and Test
- **Compile your project**: Make sure to compile your project to apply the changes.
- **Open the Editor**: After compiling, open the Unreal Editor. You should see a new category in the "Place Actors" panel named "My Custom Actors".
- **Place Your Actor**: You can now drag and drop your custom actor from the "Place Actors" panel into your level.

### 4. Optional: Add an Icon for Your Actor
If you want to add a custom icon for your actor in the "Place Actors" panel, you can modify the `FPlacementCategoryInfo` and `FPlaceableItem` to include a custom icon reference.

### Summary
In short, you extend the editor by creating a custom module that registers your actor class into a new or existing category within the "Place Actors" window. This involves a combination of creating the actor in C++ and modifying the editor's behavior through the `IPlacementModeModule` interface.

This method allows you to customize the Unreal Engine Editor to better suit your development needs, making your custom actor easily accessible for level designers.

```
